csharp/test/Google.OpenLocationCode.Tests/EncodingTests.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.OpenLocationCode.Tests
+{
+    public class EncodingTests
+    {
+        [Fact]
+        public void TestEncoding()
+        {
+            var data = LoadTestData();
+
+            foreach ( var item in data )
+            {
+                int codeLength = item.Code.Length - 1;
+
+                if ( item.Code.IndexOf( '0' ) > -1 )
+                {
+                    codeLength = item.Code.IndexOf( '0' );
+                }
+
+                var encoded = OpenLocationCode.Encode( item.Latitude, item.Longitude, codeLength )
+                    .ToString();
+
+                Assert.Equal( item.Code, encoded );
+            }
+        }
+
+        [Fact]
+        public void TestDecoding()
+        {
+            int precision = 2;
```

Comment:
Testing to two decimal places is insufficient. Change precision to 10. (And call it decimalPlaces or something.)